pkg/controller/protobuf/client/client.go

```diff
@@ -598,48 +598,62 @@ func (ctrlAdapter *controllerAdapter) Update(ctx context.Context, newResource re
 }
 
 func (ctrlAdapter *controllerAdapter) Modify(ctx context.Context, emptyResource resource.Resource, updateFunc func(resource.Resource) error) error {
+	_, err := ctrlAdapter.modify(ctx, emptyResource, updateFunc)
+
+	return err
+}
+
+func (ctrlAdapter *controllerAdapter) ModifyGetResult(ctx context.Context, emptyResource resource.Resource, updateFunc func(resource.Resource) error) (resource.Resource, error) {
```

Comment:
idk if that looks better? a suggesion

```suggestion
func (ctrlAdapter *controllerAdapter) ModifyWithResult(ctx context.Context, emptyResource resource.Resource, updateFunc func(resource.Resource) error) (resource.Resource, error) {
```

Reply:
I'm fine with that as well, let me rename